packages/ccip-server/src/index.ts

```diff
@@ -0,0 +1,234 @@
+  }
+
+  /**
+   * Add IMulticallable.multicall() support
+   * Errors (missing implemention, handler exception) are encoded as: Error(string)
+   */
+  addMulticallSupport() {
+    const selector = ethers.utils.id('Error(string)').slice(0, 10)
+    this.add(
+      [
+        'function multicall(bytes[] calldata data) external returns (bytes[] memory results)',
+      ],
+      [
+        {
+          type: 'multicall',
+          func: async (args: ethers.utils.Result, { to }: RPCCall) => [
+            await Promise.all(
+              args[0].map(async (data: any) => {
+                let error
+                try {
+                  const { status, body } = await this.call({ to, data })
+                  if (status === 200) return body.data // Q: should this be 2XX?
+                  error = body.message || 'unknown error'
+                } catch (err) {
+                  error = String(err) // Q: should this include status?
+                }
+                return ethers.utils.hexConcat([
+                  selector,
+                  ethers.utils.defaultAbiCoder.encode(['string'], [error]),
+                ])
+              }),
+            ),
+          ],
+        },
+      ],
+    )
+  }
+
+  /**
+   * Adds an interface to the gateway server, with handlers to handle some or all of its functions.
+   * @param abi The contract ABI to use. This can be in any format that ethers.js recognises, including
+   *        a 'Human Readable ABI', a JSON-format ABI, or an Ethers `Interface` object.
+   * @param handlers An array of handlers to register against this interface.
+   */
+  add(
+    abi: string | readonly (string | Fragment | JsonFragment)[] | Interface,
+    handlers: Array<HandlerDescription>,
+  ) {
+    const abiInterface = toInterface(abi)
+
+    for (const handler of handlers) {
+      const fn = abiInterface.getFunction(handler.type)
+      this.handlers[Interface.getSighash(fn)] = {
+        type: fn,
+        func: handler.func,
+      }
+    }
+  }
+
+  /**
+   * Convenience function to construct an `express` application object for the gateway.
+   * Example usage:
+   * ```javascript
+   * const ccipread = require('ccip-read');
+   * const server = new ccipread.Server();
+   * // set up server object here
+   * const app = server.makeApp('/');
+   * app.serve(8080);
+   * ```
+   * The path prefix to `makeApp` will have sender and callData arguments appended.
+   * If your server is on example.com and configured as above, the URL template to use
+   * in a smart contract would be "https://example.com/{sender}/{callData}.json".
+   * @returns An `express.Application` object configured to serve as a CCIP read gateway.
+   */
+  makeApp(
+    prefix: string,
+    ...middlewares: express.RequestHandler[]
+  ): express.Application {
+    const app = express()
+    app.use(cors())
+    app.use(express.json() as express.RequestHandler)
+    if (middlewares.length > 0) app.use(middlewares)
```

Comment:
this was the change that required the `@chainlink/ccip-read-server` to be modified allowing the gateway signature through the express' middleware pattern